dagger/hilt-node-factory-runtime/src/main/kotlin/com/bumble/appyx/dagger/hilt/node/AppyxNodeFactories.kt

```diff
@@ -0,0 +1,46 @@
+package com.bumble.appyx.dagger.hilt.node
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import dagger.hilt.EntryPoint
+import dagger.hilt.EntryPoints
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ActivityComponent
+import javax.inject.Provider
+
+@EntryPoint
+@InstallIn(ActivityComponent::class)
+internal interface AppyxNodeFactoryEntryPoint {
+    val appyxNodeFactories: AppyxNodeFactories
+}
+
+class AppyxNodeFactories(
+    private val delegate: Map<Class<*>, Provider<AppyxNodeFactory>>
+) : Map<Class<*>, Provider<AppyxNodeFactory>> by delegate
+
+inline fun <reified T : AppyxNodeFactory> AppyxNodeFactories.getAppyxNodeFactory(): T {
+    return (getValue(T::class.java) as Provider<T>).get()
+}
+
+fun appyxNodeFactories(context: Context): AppyxNodeFactories {
+    val activity = context.let {
```

Comment:
This activity finding logic could be shared with the ribs-interop module through a common library or part of core